### PR TITLE
Allow users to read license information on a node [OSF-6292]

### PR DESCRIPTION
**Pull request by brianjgeiger**
## Purpose

Allow users of the v2 API to access license information for a node and details about available licenses. (Read only)

## Changes

1. Added `/v2/licenses/` and `/v2/licenses/[license-id]` routes to show the licenses available to the user
2. Added a license relationship to the node serializer that would link to the above licenses
3. Added node_license attribute to the node serializer that would give year and copyright holder information related to the attached license
4. Ensured that withdrawn registrations do not serialize license information
5. Ensure that the license relationship can be viewed with anonymized VOL but not the node_license data
6. Allow default query on ODM Filter Views to be `None`
7. Add license read scope
8. Documentation and tests

screen shot 2016-05-16 at 7 33 33 pm

screen shot 2016-05-16 at 7 33 11 pm

## Side effects

Shouldn't be any.

## Ticket

https://openscience.atlassian.net/browse/OSF-6292

### CODE REVIEW

File: api/licenses/views.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.LICENSE_READ]
```
Comment (mfraezz) [inline api/licenses/views.py]: Note: discussed in person, these endpoints should have an `ALWAYS_PUBLIC` scope, which will be implemented in OSF-6360. Fine for now.

**Comment by mfraezz** [comment]
Please add tests to ensure "`Changes`" points 4 and 5 above:

> 4\. Ensured that withdrawn registrations do not serialize license information
> 5\. Ensure that the license relationship can be viewed with anonymized VOL but not the node_license data

Otherwise, LGTM. Back to you @brianjgeiger

**Comment by brianjgeiger** [comment]
# 4 test: there's a registration test that ensures that only specific fields are visible when the registration is withdrawn. That catches these problems automatically rather than writing individual tests which would, if someone forgot to decorate the field, also cause forget to write the test. So it's covered.

**Comment by brianjgeiger** [comment]
Number five is handled by a similar situation, where we have to define the list of things that get shown in a VOL. There's a bunch of infrastructure around it already, so I *think* testing each field individually would be not super useful, but you might check me on that.

**Comment by mfraezz** [comment]
Discussed in person, above concerns are addressed. Dropping this in RTM :shipit:

**Pull request merged**